src/test/scala/dev/kamu/cli/MetadataRepositorySpec.scala

```diff
@@ -53,4 +58,56 @@ class MetadataRepositorySpec extends FunSuite with Matchers with KamuTestBase {
       actual3 shouldEqual Seq(dsC.id, dsB.id, dsA.id)
     }
   }
+
+  test(raw"'kamu add' from HTTP") {
+    withEmptyRepo { kamu =>
+      val testDir = getRandomDir()
+      val serverPort = 8080
+      val hostPort = 9000
+      val path: Path = new Path(testDir, raw"test-dataset.yaml")
+      fileSystem.mkdirs(testDir)
+      val testServerName = s"kamu_test_http_server"
+      val testHttpServerArgs = DockerRunArgs(
+        image = "python:3.7-slim",
+        args = List("python3", "-m", "http.server", s"${serverPort}"),
+        exposePortMap = Map(hostPort -> serverPort),
+        volumeMap = Map(testDir -> new Path("/mnt/kamu")),
+        containerName = Some(testServerName),
+        detached = true
+      )
+      val testHttpServer = new DockerClient(FileSystem.get(new Configuration()))
+
+      val testHttpServerProc = new DockerProcessBuilder(
+        "http",
+        testHttpServer,
+        testHttpServerArgs
+      ).run()
+      val expected = DatasetFactory.newRootDataset()
+      kamu.metadataRepository.saveDataset(expected, path)
+      testHttpServerProc.join()
+      Thread.sleep(1000) // wait for the port to become ready, 'waitForHostPort' did work :-(
```

Comment:
Hm, I'd like to investigate this a bit further. I'm generally very much against sleeping in tests - you always either sleep to long and make tests unnecessary long, or too little and then your tests break randomly.

Reply:
that problem that I pointed out might actually be the reason why `waitForHostPort` didn't work for you

Reply:
`waitForHostPort` would wait till it becomes available, but seem to rise the exception regardless...